From c3f8aba6219dd7d0a91bb4752aa5cef1ad2e49a6 Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Tue, 23 Apr 2013 12:48:17 +0800
Subject: [PATCH 392/449] ENGR00259949-1 mxc vout:do not dqueue all bufs
 before streamoff

In ic bypass case, we cannot get a buffer dqueued before the
next 2 buffers are put on display channel, which may block
us from dequeuing the last 2 buffers at the end of the video
clip. However, we don't have to dqueue all buffers before
streaming off in the unit test. This patch changes to stream
off directly, but not to wait until all buffers are dequeued.

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
---
 test/mxc_v4l2_test/mxc_v4l2_output.c | 36 ------------------------------------
 1 file changed, 36 deletions(-)

diff --git a/test/mxc_v4l2_test/mxc_v4l2_output.c b/test/mxc_v4l2_test/mxc_v4l2_output.c
index f062e7e..3a4ffde 100644
--- a/test/mxc_v4l2_test/mxc_v4l2_output.c
+++ b/test/mxc_v4l2_test/mxc_v4l2_output.c
@@ -268,24 +268,6 @@ mxc_v4l_output_test(FILE *in)
                                         printf("v4l2_output: end of input file, g_frame_size=%d, err = %d\n", g_frame_size, err);
 					if (i <= streamon_cnt)
 						printf("v4l2_output: no display because v4l need at least %d frames\n", streamon_cnt + 1);
-					else {
-						int deq_cnt, j;
-
-						if (i < g_num_buffers)
-							deq_cnt = i;
-						else
-							deq_cnt = g_num_buffers - 1;
-						for (j=0;j<(deq_cnt - 1);j++) {
-							buf.type = V4L2_BUF_TYPE_VIDEO_OUTPUT;
-							buf.memory = g_mem_type;
-							if (ioctl(fd_v4l, VIDIOC_DQBUF, &buf) < 0)
-							{
-								printf("VIDIOC_DQBUF failed\n");
-								retval = -1;
-								break;
-							}
-						}
-					}
                                         break;
                                 }
                                 fseek(in, 0, SEEK_SET);
@@ -300,24 +282,6 @@ mxc_v4l_output_test(FILE *in)
 				if (g_loop_count-- == 0) {
 					if (i <= streamon_cnt)
 						printf("v4l2_output: no display because v4l need at least %d frames\n", streamon_cnt + 1);
-					else {
-						int deq_cnt, j;
-
-						if (i < g_num_buffers)
-							deq_cnt = i;
-						else
-							deq_cnt = g_num_buffers - 1;
-						for (j=0;j<(deq_cnt - 1);j++) {
-							buf.type = V4L2_BUF_TYPE_VIDEO_OUTPUT;
-							buf.memory = g_mem_type;
-							if (ioctl(fd_v4l, VIDIOC_DQBUF, &buf) < 0)
-							{
-								printf("VIDIOC_DQBUF failed\n");
-								retval = -1;
-								break;
-							}
-						}
-					}
 					break;
 				}
 				count = 30;
-- 
1.8.0

